experiments/experiment_071224_rw_default_no_coupling.py

```diff
@@ -73,57 +78,81 @@ def objective(trial):
             "resource_cluster_radius": RESOURCE_CLUSTER_RADIUS,
             "keep_overall_abundance": True,
         },
-        iterations= NUM_ITERATIONS,
+        iterations=NUM_ITERATIONS,
         number_processes=None,  # use all CPUs
         max_steps=max_sim_steps,
         data_collection_period=-1,  # only the last step
     )
-    results = pd.DataFrame(results) ### MAKE THIS SCALE FREE BY NORMALISING WITH NUM OF STEPS
+    results = pd.DataFrame(
+        results
+    )  ### MAKE THIS SCALE FREE BY NORMALISING WITH NUM OF STEPS
 
     # Filter out agents (resource AgentID is usually 0, so we remove it)
     mask = results.AgentID != 0
 
-    # Calculate the average collected resource
-    avg_collected_resource = results.loc[mask, "collected_resource"].mean()
+    agent_metrics = (
+        results.groupby("AgentID")
+        .agg(
+            {
+                "collected_resource": "max",
+                "traveled_distance": "max",
+                "time_to_first_catch": "first",
+            }
+        )
+        .reset_index()
+    )
 
-    return avg_collected_resource
+    # Calculate efficiency (resource/distance)
+    agent_metrics["efficiency"] = agent_metrics["collected_resource"] / (
+        agent_metrics["traveled_distance"] + 1e-10
+    )
+
+    # Use mean efficiency as objective
+    avg_efficiency = agent_metrics["efficiency"].mean()
+
+    return avg_efficiency
 
 
 if __name__ == "__main__":
     # Create the Optuna study and optimize the objective function
-    study = optuna.create_study(direction="maximize")
-    study.optimize(objective, n_trials=50, n_jobs=-1)
+    study_name = "foraging-db"  # Unique identifier of the study
+    storage_name = "sqlite:///{}.db".format(study_name)
+
+    study = optuna.create_study(direction="maximize", storage=storage_name)
+
+    study.optimize(objective, n_trials=5, n_jobs=1)
 
     # Print the best trial results
     trial = study.best_trial
     print("Average collected resource: {}".format(trial.value))
     print("Best hyperparameters: {}".format(trial.params))
 
-    # Visualization: Plot the optimization history
-    import plotly.io as pio
+    if plotting:
```

Comment:
For this and all other experiments, all code below this line should not be in this file